ui/src/components/wallettransfer.jsx

```diff
@@ -16,8 +16,8 @@ const Wallettransfer = ({ clientConfig, contractState, setConfirmationModal, set
 
     function transferWalletOwnerShip() {
         openContractCall({
-            contractAddress: userAddress,
-            contractName,
+            contractAddress: sharedAddress.split('.')[0] || userAddress,
```

Comment:
better to split only once like
const [contractAddress, contractName] = sharedAddress.split('.')

Reply:
@Zeus-Adin Could you address this as well?